near-contract-standards/src/non_fungible_token/enumeration/enumeration_impl.rs

```diff
@@ -3,16 +3,21 @@ use crate::non_fungible_token::token::Token;
 use crate::non_fungible_token::NonFungibleToken;
 use near_sdk::json_types::U128;
 use near_sdk::{env, require, AccountId};
+use std::collections::HashMap;
 
 type TokenId = String;
 
 impl NonFungibleToken {
     /// Helper function used by a enumerations methods
     /// Note: this method is not exposed publicly to end users
     fn enum_get_token(&self, owner_id: AccountId, token_id: TokenId) -> Token {
-        let metadata = self.token_metadata_by_id.as_ref().unwrap().get(&token_id);
-        let approved_account_ids =
-            Some(self.approvals_by_id.as_ref().unwrap().get(&token_id).unwrap_or_default());
+        let metadata = self
+            .token_metadata_by_id
+            .as_ref()
+            .and_then(|token_metadata_by_id| token_metadata_by_id.get(&token_id));
+        let approved_account_ids = self.approvals_by_id.as_ref().and_then(|approvals_by_id| {
+            approvals_by_id.get(&token_id.to_string()).or_else(|| Some(HashMap::new()))
```

Comment:
```suggestion
            approvals_by_id.get(&token_id))
```
I think that if approval standard isn't implemented this should be `None` on the token. @BenKurrek can you confirm?